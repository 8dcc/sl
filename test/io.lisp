;;------------------------------------------------------------------------------
;; Features tested in this source:
;;   - Input primitives: `read', `read-str'
;;   - Output primitives: `write', `print-str'
;;------------------------------------------------------------------------------

(define writeln
    (lambda (e)
      (write e)
      (print-str "\n")
      tru))

(begin
 (print-str "Reading two expressions...\n")
 (writeln (read))
 (writeln (read)))

(begin
 (print-str "\nWriting two expressions...\n")
 (writeln '(+ 1 2 3 (f 'sym "str")))
 (writeln writeln))

(begin
 (print-str "\nReading two user-strings...\n")
 (writeln (read-str))
 (writeln (read-str ".\n")))
