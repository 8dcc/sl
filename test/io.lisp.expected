Standard library loaded.
<lambda>
Reading two expressions...
123
(+ 1 2 3 (- 5 4))
tru

Writing two expressions...
(+ 1 2 3 (f (quote sym) "str"))
(lambda (e) (write e) (print-str "\n") tru)
tru

Scanning two user strings...
"User string..."
"Another delimited line"
tru
